tests/unit/WidgetBase.ts

```diff
@@ -20,1031 +20,1071 @@ function testDecorator(func?: Function) {
+			},
+			'render with nested children'() {
+				class TestWidget extends WidgetBase<any> {
+					render() {
+						return v('div', [
+							v('header', [
+								v('section')
+							])
+						]);
+					}
 				}
-			}
 
-			const widget: any = new TestWidget();
-			const result = <VNode> widget.__render__();
-			assert.lengthOf(result.children, 1);
-			assert.strictEqual(result.children![0].vnodeSelector, 'header');
-			assert.strictEqual(result.children![0].children![0].vnodeSelector, 'section');
-		},
-		'render with a text node children'() {
-			class TestWidget extends WidgetBase<any> {
-				render() {
-					return v('div', [ 'I am a text node' ]);
+				const widget: any = new TestWidget();
+				const result = <VNode> widget.__render__();
+				assert.lengthOf(result.children, 1);
+				assert.strictEqual(result.children![ 0 ].vnodeSelector, 'header');
+				assert.strictEqual(result.children![ 0 ].children![ 0 ].vnodeSelector, 'section');
+			},
+			'render with a text node children'() {
+				class TestWidget extends WidgetBase<any> {
+					render() {
+						return v('div', [ 'I am a text node' ]);
+					}
 				}
-			}
 
-			const widget: any = new TestWidget();
-			const result = <VNode> widget.__render__();
-			assert.isUndefined(result.children);
-			assert.equal(result.text, 'I am a text node');
-		},
-		'render returns array'() {
-			class TestChildWidget extends WidgetBase {
-				render() {
-					return [
-						v('div', [ 'text' ]),
-						v('span', { key: 'span' })
-					];
+				const widget: any = new TestWidget();
+				const result = <VNode> widget.__render__();
+				assert.isUndefined(result.children);
+				assert.equal(result.text, 'I am a text node');
+			},
+			'render returns array'() {
+				class TestChildWidget extends WidgetBase {
+					render() {
+						return [
+							v('div', [ 'text' ]),
+							v('span', { key: 'span' })
+						];
+					}
 				}
-			}
 
-			class TestWidget extends WidgetBase {
-				render() {
-					return v('div', [ w(TestChildWidget, {}) ]);
+				class TestWidget extends WidgetBase {
+					render() {
+						return v('div', [ w(TestChildWidget, {}) ]);
+					}
 				}
-			}
 
-			const widget = new TestWidget();
-			const result: any = widget.__render__();
-			assert.strictEqual(result.vnodeSelector, 'div');
-			assert.lengthOf(result.children, 2);
-			assert.strictEqual(result.children![0].vnodeSelector, 'div');
-			assert.strictEqual(result.children![0].text, 'text');
-			assert.strictEqual(result.children![1].vnodeSelector, 'span');
-			assert.strictEqual(result.children![1].properties.key, 'span');
-		},
-		'instance gets passed to VNodeProperties as bind to widget and all children'() {
+				const widget = new TestWidget();
+				const result: any = widget.__render__();
+				assert.strictEqual(result.vnodeSelector, 'div');
+				assert.lengthOf(result.children, 2);
+				assert.strictEqual(result.children![ 0 ].vnodeSelector, 'div');
```

Comment:
Not sure we specifically pad indexes of arrays like this?

Reply:
I updated my IDE to not add the spaces and reformatted the files.

Reply:
I updated my IDE settings and reformatted the files.